### PR TITLE
Make it possible to hide the menu bar

### LINKED ISSUE #188: visual improvement: hide menu bar
after a week of using the file manager, i usually have set up everything the way i like it. then, i have no need for the menu bar anymore. i would like to be able to hide it for a cleaner look.

thunar and other file managers can hide (and unhide) their menu bars with a keyboard shortcut (usually: CTRL + M). they usually have an option in the "view" menu to hide the menu bar, too.

### PR DESCRIPTION
**Pull request by npmiller**
Fixes #188.

### CODE REVIEW

File: pcmanfm/mainwindow.cpp [hunk 1]
```diff
@@ -260,6 +267,28 @@ void MainWindow::addTab(FmPath* path) {
   }
 }
 
+void MainWindow::toggleMenuBar() {
+  Settings& settings = static_cast<Application*>(qApp)->settings();
+  bool showMenuBar = !settings.showMenuBar();
+
+  if (!showMenuBar) {
+    QScopedPointer<QMessageBox> warning(
```
Comment (palinek) [inline pcmanfm/mainwindow.cpp]: Why are you creating the `QMessageBox` widget on the heap and not on the the stack?

File: pcmanfm/mainwindow.cpp [hunk 2]
```diff
@@ -260,6 +267,28 @@ void MainWindow::addTab(FmPath* path) {
...
+
+    warning->show();
+    warning->exec();
+    if (QMessageBox::Cancel == warning->result()) {
+      return;
+    }
+  }
+
+  ui.menubar->setVisible(showMenuBar);
+  settings.setShowMenuBar(showMenuBar);
```
Comment (palinek) [inline pcmanfm/mainwindow.cpp]: Isn't the `setShowMenuBar` superfluous?

Comment (npmiller) [inline pcmanfm/mainwindow.cpp]: No, it's to update the settings so that the state of the menu bar will be remembered.

Comment (palinek) [inline pcmanfm/mainwindow.cpp]: Sorry, my fault.. I misinterpreted the code...

**Comment by PCMan** [comment]
Looks good. Wouldn't it be simpler if you use QMessageBox::warning() directly?
http://doc.qt.io/qt-5/qmessagebox.html#warning
Thanks!

**Comment by npmiller** [comment]
Good point, updated to use `QMessageBox::warning` instead.

**Comment by PCMan** [comment]
Merged. Thanks!

**Pull request closed without merging**